torch/csrc/distributed/rpc/request_callback_impl.cpp

```diff
@@ -129,9 +121,48 @@ std::shared_ptr<FutureMessage> RequestCallbackImpl::processRpc(
         // addForkOfOwner as it is not a fork. To allow callee to distinguish
         // when this request is sent to self, the caller will set forkId using
         // rrefId (OwnerRRef does not have a forkId anyway).
-        ctx.addForkOfOwner(rrefId, forkId);
+        RRefContext::getInstance().addForkOfOwner(rrefId, forkId);
       }
-      return wrap(RemoteRet(rrefId, forkId).toMessage());
+
+      auto onCompletion =
+          [ownerRRef, rrefId, forkId, messageId](IValue val) -> Message {
+        ownerRRef->setValue(std::move(val));
+        Message m = RemoteRet(rrefId, forkId).toMessage();
+        m.setId(messageId);
+        return m;
+      };
+
+      // scriptRemoteCall is only alive within this block, use reference to
+      // avoid copy
+      auto& stack = scriptRemoteCall.stackRef();
+      if (scriptRemoteCall.hasOp()) {
+        scriptRemoteCall.op()->getOperation()(stack);
+        TORCH_INTERNAL_ASSERT(
+            stack.size() == 1,
+            "Return value of a builtin operator or a "
+            "TorchScript function should be a single IValue, got a vector of "
+            "size ",
```

Comment:
this is prior to this PR, but can we merge these three lines into two?

Reply:
done